device/pcie/pci_device.cpp

```diff
@@ -493,3 +497,26 @@ void PCIDevice::write_tlb_reg(uint32_t byte_addr, std::uint64_t value_lower, std
     }
     tt_driver_atomics::mfence(); // Otherwise subsequent WC loads move earlier than the above UC store to the TLB register.
 }
+
+bool PCIDevice::is_hardware_hung() {
+    volatile const void *addr = reinterpret_cast<const char *>(bar0_uc) + (get_architecture_implementation()->get_arc_reset_scratch_offset() + 6 * 4) - bar0_uc_offset;
```

Comment:
nit: space between char and *

Reply:
I'm sorry, I specifically don't want to fix comments like since adding clang formatting to our repo will fix all of these automatically :) Will work on adding that soon.